### PR TITLE
merge stable into master

### PR DESCRIPTION
**Pull request by KevinCorcoran**
This required integrating the comidi refactor on top of the changes to
the Master's and CA's HTTP APIs which had already landed in master.
This was difficult.

### CODE REVIEW

**Comment by KevinCorcoran** [comment]
I just added another commit to this PR.  It should fix the Travis build.

**Comment by KevinCorcoran** [comment]
This can probably be merged despite the Travis failure.  Looks like it's due to memory usage.

**Comment by KevinCorcoran** [comment]
Additionally, I've gotten green test runs on my local machine.

File: README_BRANCHING.md [hunk 1]
```diff
@@ -28,39 +23,30 @@ should be considered roughly equivalent to the Puppet Server 1.0.2 release in
...
+hopefully close to one another.  These will be the next major feature releases
+(as opposed to 2.0 and 2.1, which are mostly just compatibility releases), and 
+will contain several new features, tuning improvements, etc.
 
-We'll update this document to reflect changes to that restriction as things
-progress.
+As development toward the 1.1 and 2.1 releases proceeds, changes from the
+`stable` and `2.1.x` branches, respectively, should be merged up to the `master`
+branch.  Any changes that may be considered too risky or inappropriate to be
+included in either the 1.1 or 2.1 releases could be targeted at the `master`
```
Comment (haus) [inline README_BRANCHING.md]: could => should?

**Comment by rlinehan** [comment]
:+1: looks good to me.

File: README_BRANCHING.md (hunk 1 above)
Comment (KevinCorcoran) [inline README_BRANCHING.md]: coulda, shoulda, woulda

**Comment by dankreek** [comment]
:+1: Great merge! 5/5 Would review again!

**Pull request merged**